Esri/arcgis-maps-sdk-swift-samples
# [Fix] Manage operational layers bugs

## Description

This PR implements fixes for two bugs found with the `Manage operational layers` sample:
1. The Manage Layers settings pane showed as a blank screen when the it was accessed on Mac Catalyst. This was fixed by removing the `NaviagtionView` since it would cause a blank screen when used with the custom `sheet` implementation found in the Sample Viewer. This required adding UI components to mimic the original UI design. 
2. On iOS 15, the layer list rows couldn't be reordered by holding and dragging. This was fixed by replacing the "Done" button with an "Edit" button that will allow users to reorder the list items with the drag handles.
    - Note: The custom minus button implementation was kept because, without it, there is a bug where you are not able to remove list items that were removed and then re-added to the list. Also, the plus and minus buttons are different sizes on Mac Catalyst when the default implementation is used. Using the custom minus button does cause the row content on Mac Catalyst and iOS 15 to move right when the "Edit" button is pressed, but that cannot be helped. 

## Linked Issue(s)

- `swift/issues/4355`

## How To Test

- Tap the "Manage Layers" button.
- Tap "Edit" and drag the handles to reorder the operational layers.
- Tap "-" or swipe to remove an operational layer.
- Tap "+" to add a removed layer.
- Ensure it works on Mac Catalyst and iOS 15. 

## Screenshots

|Before|After|
|:-:|:-:|
|![Screen Shot, Before](https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/34120440/b54a1a21-8813-44ef-9360-6a654cbfa1a4)|![Screen Shot, After](https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/34120440/78501ec9-a789-4710-abd8-22ea3593283e)|
